planner/core/planbuilder.go

```diff
@@ -3721,15 +3721,38 @@ func (b *PlanBuilder) buildInsert(ctx context.Context, insert *ast.InsertStmt) (
 	mockTablePlan.SetSchema(insertPlan.tableSchema)
 	mockTablePlan.names = insertPlan.tableColNames
 
-	checkRefColumn := func(n ast.Node) ast.Node {
+	checkRefColumn := func(n ast.Node) (ast.Node, error) {
 		if insertPlan.NeedFillDefaultValue {
-			return n
+			return n, nil
 		}
 		switch n.(type) {
 		case *ast.ColumnName, *ast.ColumnNameExpr:
 			insertPlan.NeedFillDefaultValue = true
+		case *ast.SubqueryExpr:
+			sel := n.(*ast.SubqueryExpr).Query
+			switch sel.(type) {
+			case *ast.SelectStmt:
+				if sel.(*ast.SelectStmt).From != nil {
+					left := sel.(*ast.SelectStmt).From.TableRefs.Left
+					switch left.(type) {
+					case *ast.TableSource:
```

Comment:
Sorry for the late reply. It seems that we need to also handle the following case:
```
mysql> INSERT INTO t1 (x) VALUES ((SELECT t2.x FROM t2 join t1 on t1.x = t2.x));
ERROR 1093 (HY000): You can't specify target table 't1' for update in FROM clause
```

Reply:
Maybe we can recursively extract all the tables in `TableRefs`:
1. If it is a `Join` node, then recursively look at the left node and the right node.
2. Otherwise, if it is a `TableSource` node, check its table name.

Reply:
Thks. Done as guided, please check. BR

Reply:
/cc @xuyifangreeneyes